### PR TITLE
Add documentation for upgrading using DAML script and triggers

### PR DESCRIPTION
**Pull request by cocreature**
This PR adds documentation for how you can automate the upgrade
process from the Coin example using DAML Script and DAML Triggers.

Having written this, I think we might want to add some integration
tests for this but it’s rather complex to setup since it ties together
so many pieces so I’d like to leave it out for now. We do at least
test that the code compiles and we all knows that code that compiles
never has bugs.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: docs/source/upgrade/automation.rst [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+.. _upgrade-automation:
+
+Automating the Upgrade Process
+##############################
+
+In this section, we are going to automate the upgrade of our coin
+process using :doc:`DAML Script</daml-script/index>` and :doc:`DAML
+Triggers </triggers/index>`. Note that automation for upgrades just
+like the upgrade models is specific to an individual
+application. Nevertheless, we have found that the pattern shown here
```
Comment (associahedron) [inline docs/source/upgrade/automation.rst:13]: Wording suggestion:

```suggestion
Triggers </triggers/index>`. Note that automation for upgrades is
specific to an individual application, just like the upgrade models.
Nevertheless, we have found that the pattern shown here
```

File: docs/source/upgrade/automation.rst [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+the upgrade:
+
+#. A DAML script that will be executed once by Alice and creates an
+   ``UpgradeCoinProposal`` contract for each owner.
+#. Navigator to accept the ``UpgradeCoinProposal`` as Bob. While we
+   could also use a DAML script to accept the proposal, this step will
+   often be exposed as part of a web UI so doing it interactively in
+   Navigator resembles that workflow more closely.
+#. A long-running DAML trigger that upgrades all ``Coin`` contracts
+   for which there is a corresponding ``UpgradeCoinAgreeement``.
```
Comment (associahedron) [inline docs/source/upgrade/automation.rst:48]:
```suggestion
   for which there is a corresponding ``UpgradeCoinAgreement``.
```

File: docs/source/upgrade/automation.rst [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+Implementation of the DAML Script
+=================================
+
+In our DAML Script, we are first going to query the ACS to find all
```
Comment (associahedron) [inline docs/source/upgrade/automation.rst:53]: Possible suggestion:

```suggestion
In our DAML Script, we are first going to query the ACS (Active Contract Set) to find all
```

File: docs/source/upgrade/automation.rst (hunk 2 above)
Comment (cocreature) [reply docs/source/upgrade/automation.rst:48]: It took me 3 times until I spotted the difference :slightly_smiling_face:

**Pull request merged**